Misc fixes

mr-miles opened the pull request:
Fixes for a few things we've been bitten by recently:
- AbstractBootstrapper's UseConfiguration method failed with a NullReference exception, fixed
- Consistent handling of "localhost" urls at service startup and in the endpointrouter
  startup replaces localhost with the machine name, which then means endpointrouter entries
  for localhost don't get routed correctly.
- Castle container's component registration method moved to an extension method for use
  outside the bootstrapper
- Tests

mr-miles committed:
Bug fix and extension method

+ Fix bug in AbstractBootStrapper where ability to provide a config
section (other than the default) was broken.
+ Moved castle consumer registration into an extension method for use
outside the bootstrapper without copy-and-pasting.

mr-miles committed:
Consistent handling of localhost

Fix an issue where queue URLs containing localhost were being normalised
to the machine name when the bus was set up, but the same was not being
done in the EndpointRouter.

mr-miles committed:
Fix up

Fix typos and other issues

mr-miles committed:
Update Rhino.ServiceBus.Tests/Rhino.ServiceBus.Tests.csproj

Add endpoint router tests

CoreyKaylor commented:
There's one Autofac related test that fails from these changes. Can you take a look?

mr-miles committed:
Ordering possible method calls

Support ordering whereby DefaultHost.InitializeContainer calls InitializeContainer() BEFORE UseConfiguration()

mr-miles commented:
Done - that test now passes

CoreyKaylor commented:
Thanks, closed with 02c66b6e4fe7bbc7
